drivers/crypto/crypto_mchp_sha_g1.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (C) 2025 Microchip Technology Inc. and its subsidiaries
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#define DT_DRV_COMPAT microchip_sha_g1_crypto
+
+#define LOG_LEVEL CONFIG_CRYPTO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(cypto_mchp_sha_g1);
+
+#include <zephyr/crypto/crypto.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control/mchp_sam_pmc.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/atomic.h>
+
+#define MCHP_SHA_CAPS_SUPPORT (CAP_INPLACE_OPS | CAP_SEPARATE_IO_BUFS |		\
+			       CAP_SYNC_OPS | CAP_NO_IV_PREFIX)
+
+struct crypto_mchp_sha_cfg {
+	sha_registers_t *const regs;
+	const struct sam_clk_cfg clock_cfg;
+};
+
+struct crypto_mchp_sha_data {
+	struct k_sem device_sem;
+};
+
+struct crypto_mchp_sha_session {
+	struct crypto_mchp_sha_algo_cfg const *algo_cfg;
+	bool in_use;
+};
+
+static const struct crypto_mchp_sha_algo_cfg {
+	enum hash_algo algo;
+	uint32_t sha_mr_algo;
+	uint8_t dgst_len;
+	uint8_t block_size;
+} mchp_sha_algo_cfgs[4] = {
+	{CRYPTO_HASH_ALGO_SHA224, SHA_MR_ALGO_SHA224, 224/8, 64},
+	{CRYPTO_HASH_ALGO_SHA256, SHA_MR_ALGO_SHA256, 256/8, 64},
+	{CRYPTO_HASH_ALGO_SHA384, SHA_MR_ALGO_SHA384, 384/8, 128},
+	{CRYPTO_HASH_ALGO_SHA512, SHA_MR_ALGO_SHA512, 512/8, 128},
+};
```

Comment:
Let's not mix struct definition with instantiation.
Define the instantiation separate.